### DISCUSSION TITLE
Are custom protocol activation docs uptodate?

### DISCUSSION THREAD

**Discussion by ahmed-abdelrazek** [unoplatform/uno#10558] [Q&A] [answered]
i was following this page https://platform.uno/docs/articles/features/protocol-activation.html so i can open my app from the browser until i reached to 
`protected override void OnActivated(IActivatedEventArgs e)`
which uno tells me it doesn't exist when i try to build
![image](https://user-images.githubusercontent.com/7858946/203824531-58fdc78d-9dc7-4557-a6a0-8391a9c9ee68.png)

when i try to get the arguments from `OnLaunched` it always turn out empty string
![image](https://user-images.githubusercontent.com/7858946/203825435-0b516530-4930-4dd9-b5ae-9dc49ccf04e8.png)

![image](https://user-images.githubusercontent.com/7858946/203825544-6ae0a146-9ecc-40ae-bf1e-276a12162acf.png)

and i try to open the app like this from the browser with url  `cycoau:youtube` and it does open but with empty arguments shouldn't it be `youtube` or i'm missundertanding something

**Comment by jeromelaban** [comment]
@MartinZikmund

**Comment by ahmed-abdelrazek** [comment, marked as answer]
any news?!

**Reply by MartinZikmund** [reply]
Sorry for the delay. Unfortunately in case of Windows App SDK the approach to protocol activation is different, so it has to be implemented separately - see [docs](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/applifecycle/applifecycle-rich-activation).

Before we bring this approach to Uno, I recommend wrapping the `OnActivated` method in `#if HAS_UNO` which will ensure this part of code will compile only against Uno Platform targets, not Windows. Conversely you can use `#if !HAS_UNO` to implement Windows-specific behavior as needed

**Reply by ahmed-abdelrazek** [reply]
no problem, and thanks for investigating this

but do you mean the other way around?! as far as i know `OnActivated` is part of UWP/WinUI
or do you mean to use it as you suggested with other platforms and implement this [docs](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/applifecycle/applifecycle-rich-activation) just for the UWP

aslo will it work with WPF on windows 7 and Linux?!

**Reply by MartinZikmund** [reply]
It is part of UWP, but not WinUI 3 (aka Windows App SDK) (I am guessing because it would not be possible to support it in unpackaged apps?)
<img width="256" alt="image" src="https://user-images.githubusercontent.com/1075116/211188802-c3e14868-d15c-4834-956e-15b72dac054b.png">

You can use `OnActivated` as described in the docs for Uno targets, but not for Windows target (which is not UWP, but Windows App SDK).

For WPF - I think we don't support this yet, but will create an issue to look into protocol activation there

**Reply by ahmed-abdelrazek** [reply]
wpf and linux i think will be more complicated
wpf needs registry edit to add your custom protocol to windows and link it to the app

**Comment by ahmed-abdelrazek** [comment]
https://github.com/unoplatform/uno/issues/10915
